tensorflow/python/distribute/input_lib_test.py

```diff
@@ -1250,5 +1250,55 @@ def sum_while_loop(ds):
     self.assertAllEqual(nest.flatten(sums), [expected_for_sum] * 3)
 
 
+class DistributedIteratorPerReplicaTest(test.TestCase, parameterized.TestCase):
+
+  @combinations.generate(
+      combinations.combine(
+          mode=["eager"],
+          distribution=[
+              strategy_combinations.mirrored_strategy_with_two_gpus,
+              strategy_combinations.mirrored_strategy_with_cpu_1_and_2,
+              strategy_combinations.mirrored_strategy_with_gpu_and_cpu,
+              ]))
+
+  def setUp(self):
+    super(DistributedIteratorPerReplicaTest, self).setUp()
+    strategy_combinations.set_virtual_cpus_to_at_least(3)
+
+  def testInputSignatureForPerReplicaValues(self, distribution):
+    def dataset_fn(input_context):
+      return tf.data.Dataset.from_tensor_slices(
+          np.array(np.arange(0, 16)).reshape(4, 4))
+
+    input_options = tf.distribute.InputOptions(
+        replication_mode=tf.distribute.InputReplicationMode.PER_REPLICA)
```

Comment:
Can we also add tests for copy_to_device = False and prefetch_to_device = False cases?  You can use parameterization if that helps.